bindings/python/pymongocrypt/crypto.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2019-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Internal crypto callbacks for libmongocrypt."""
+
+import os
+import traceback
+
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+from cryptography.hazmat.primitives.hashes import Hash, SHA256, SHA512
+from cryptography.hazmat.primitives.hmac import HMAC
+from cryptography.hazmat.backends import default_backend
+
+from pymongocrypt.binary import _to_bytes, _write_bytes
+from pymongocrypt.binding import ffi, lib
+from pymongocrypt.compat import str_to_bytes
+
+
+def _callback_error_handler(exception, exc_value, tb):
+    """Set the mongocrypt_status_t on error."""
+    # From cffi docs: "First check if traceback is not None (it is None e.g.
+    # if the whole function ran successfully but there was an error converting
+    # the value returned: this occurs after the call)."
+    if tb is None:
+        return
+
+    status = tb.tb_frame.f_locals['status']
+    msg = str_to_bytes(''.join(traceback.format_exception(
+        exception, exc_value, tb)))
+    # TODO: the +1 in len(msg)+1 is a libmongocrypt bug.
+    lib.mongocrypt_status_set(
+        status, lib.MONGOCRYPT_STATUS_ERROR_CLIENT, 1, msg, len(msg)+1)
+
+    return False
+
+
+@ffi.callback(
+    "bool(void *, mongocrypt_binary_t *, mongocrypt_binary_t *,"
+    "     mongocrypt_binary_t *, mongocrypt_binary_t *, uint32_t *,"
+    "     mongocrypt_status_t *)",
+    onerror=_callback_error_handler)
+def aes_256_cbc_encrypt(ctx, key, iv, input, output, bytes_written, status):
+    backend = default_backend()
+    # TODO: Java uses noPadding but CBC says padding is required:
+    # https://cryptography.io/en/latest/hazmat/primitives/symmetric-encryption/#cryptography.hazmat.primitives.ciphers.modes.CBC
```

Comment:
@jyemin This comment is referencing the Java bindings:
https://github.com/mongodb/libmongocrypt/blob/5a2ff2d72c255b0f291891880b5163a8f18ab486/bindings/java/mongocrypt/src/main/java/com/mongodb/crypt/capi/MongoCryptImpl.java#L105-L108

Is that a bug in the Java bindings?

Reply:
Jeff says:
> Its not a bug.  The library does the padding.

Reply:
Specifically, libmongocrypt is doing the padding. We (or cryptography) can't do further padding on top of that. You should add a comment in our code making the situation clear. I'd hate to have a security researcher opening a security issue over a non-issue.

Reply:
Done.